File: leetcode3/최민지/315-Count-of-Smaller-Numbers-After-Self.py
class Solution:
    def countSmaller(self, nums: List[int]) -> List[int]:
        result = []
        for i in range(len(nums)):
            temp = 0
            for j in range(i+1, len(nums)):
                if nums[j] < nums[i]:
                    temp += 1
            result.append(temp)

        return result

        

File: leetcode3/황은지/315. Count of Smaller Numbers After Self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var countSmaller = function (nums) {
  const output = Array(nums.length);
  // 이렇게 하니까 잘못 참조 함..
  for (let i = nums.length - 1; i >= 0; i--) {
    output[i] = 0;
    for (let j = i + 1; j < nums.length; j++) {
      if (nums[j] <= nums[i]) {
        if (nums[j] === nums[i]) output[i] = output[j];
        else output[i] = output[j] + 1;
        break;
      }
    }
  }

  return output;
};
